src/state/slices/portfolioSlice/portfolioSlice.ts

```diff
@@ -161,8 +165,19 @@ export const accountToPortfolio: AccountToPortfolio = args => {
         portfolio.accounts.byId[accountSpecifier].push(caip19)
         portfolio.accounts.ids.push(accountSpecifier)
 
-        portfolio.assetBalances.byId[caip19] = ethAccount.balance
-        portfolio.assetBalances.ids.push(caip19)
+        // if asset already exist inside assetBalances, add balance to existing balance
+        if (!portfolio.assetBalances.byId[caip19]) {
+          portfolio.assetBalances.byId[caip19] = ethAccount.balance
+        } else {
+          portfolio.assetBalances.byId[caip19] = sumBalance(
```

Comment:
we can simply this to just the else clause and always sum it by doing `sumBalance(portfolio.assetBalances.byId[caip19] ?? '0', ethAccount.balance`